app/src/main/kotlin/com/softteco/template/ui/feature/forgotPassword/ForgotPasswordScreen.kt

```diff
@@ -0,0 +1,134 @@
+private fun ScreenContent(
+    state: ForgotPasswordViewModel.State,
+    modifier: Modifier = Modifier,
+    onBackClicked: () -> Unit = {}
+) {
+    TextSnackbarContainer(
+        modifier = modifier,
+        snackbarText = stringResource(state.snackBar.textId),
+        showSnackbar = state.snackBar.show,
+        onDismissSnackbar = state.dismissSnackBar,
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.spacedBy(Dimens.PaddingExtraLarge),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            CustomTopAppBar(
+                stringResource(id = R.string.forgot_password),
+                showBackIcon = true,
+                modifier = Modifier.fillMaxWidth(),
+                onBackClicked = onBackClicked
+            )
+            Column(
+                modifier = Modifier.padding(Dimens.PaddingNormal),
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                if (state.loading) {
+                    Text(stringResource(id = R.string.loading))
+                }
+                EmailField(state = state, Modifier.fillMaxWidth())
+                Box(
+                    modifier = Modifier.padding(
+                        Dimens.PaddingNormal,
+                        Dimens.PaddingLarge,
+                        Dimens.PaddingNormal
+                    )
+                ) {
+                    Button(
+                        onClick = {
+                            state.onRestorePasswordClicked()
+                        },
+                        shape = MaterialTheme.shapes.large,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .height(Dimens.PaddingExtraLarge)
+
+                    ) {
+                        Text(text = stringResource(id = R.string.restore_password))
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun EmailField(
+    state: ForgotPasswordViewModel.State,
+    modifier: Modifier = Modifier,
+) {
+    Column {
+        OutlinedTextField(
+            value = state.emailValue,
+            onValueChange = {
+                state.onEmailChanged(it)
+            },
+            modifier = modifier,
+            label = {
+                Text(text = stringResource(id = R.string.email))
+            },
+            isError = state.fieldStateEmail is EmailFieldState.Empty || state.fieldStateEmail is EmailFieldState.Error
+        )
+        val errorText =
+            when (state.fieldStateEmail) {
+                is EmailFieldState.Empty -> stringResource(R.string.required)
+                is EmailFieldState.Error -> stringResource(R.string.email_not_valid)
+                EmailFieldState.Success -> ""
+                is EmailFieldState.Waiting -> ""
+            }
```

Comment:
can be simplified:

when (state.fieldStateEmail) {
                is EmailFieldState.Empty -> stringResource(R.string.required)
                is EmailFieldState.Error -> stringResource(R.string.email_not_valid)
                else -> ""
            }

Reply:
fixed